tsdb/exemplar.go

```diff
@@ -274,6 +275,42 @@ func (ce *CircularExemplarStorage) validateExemplar(key []byte, e exemplar.Exemp
 	return nil
 }
 
+// outOfOrderInsertionPoint determines the insertion point for an out of order exemplar.
+// If it should instead turn out to be a duplicate, an error is returned.
+func (ce *CircularExemplarStorage) outOfOrderInsertionPoint(key []byte, e exemplar.Exemplar) (int, error) {
```

Comment:
We'll have to prepare for receiving multiple exemplars per sample for native histograms. For example after every scrape we can get potentially 10 exemplars. Even if the exemplars don't change, with this algorithm we'll do a 10 searches. So I feel like we need some kind of trick to deal with this.